completion.go

```diff
@@ -57,14 +54,103 @@ func completionCommandAction(ctx context.Context, cmd *Command) error {
 	}
 	s := cmd.Args().First()
 
-	if rc, ok := shellCompletions[s]; !ok {
+	renderCompletion, ok := shellCompletions[s]
+	if !ok {
 		return Exit(fmt.Sprintf("unknown shell %s, available shells are %+v", s, shells), 1)
-	} else if c, err := rc(cmd); err != nil {
+	}
+
+	completionScript, err := renderCompletion(cmd, appName)
+	if err != nil {
+		return Exit(err, 1)
+	}
+
+	_, err = cmd.Writer.Write([]byte(completionScript))
+	if err != nil {
 		return Exit(err, 1)
-	} else {
-		if _, err = cmd.Writer.Write([]byte(c)); err != nil {
-			return Exit(err, 1)
-		}
 	}
+
 	return nil
 }
+
+func genBashCompletion(appName string) string {
```

Comment:
can you provide the reason why you moved away from embedFs to having the completion inline with the code ? In the previous approach it was very easy to lookup where the autcompletes were and what they were doing

Reply:
This is to make it easier to use `%[1]s` since then declaration and usage is in the same place.

Reply:
I'm not sure about this either. We get nice syntax highlighting from the code editor/IDE when the autocomplete is in a separate file. But when it's inlined, it's treated as a string, which makes it less readable.

Reply:
I wouldn't say it's a big problem. Characters like `%[1]` aren't valid shell function names anyway and will probably be highlighted wrongly.

I also think keeping everything related to completion in a single file is quote convenient.

Reply:
> I also think keeping everything related to completion in a single file is quote convenient.

Scrolling back and forth between template and its code doesn't seem convenient to me. It is better to have two panes - one with template and another with content.

Highlighting of template variables as errors could actually be a  feature.

Reply:
GoLand highlights when too many or too few arguments are used in the `fmt.Sprintf` call. I think this is already a nice benefit.

I may move scripts into Go code in separate PR.